### PR TITLE
starting report templates to save time down the road

### PR DESCRIPTION
**Pull request by regineheberlein**

### CODE REVIEW

File: reports/plain_vanilla_report_templates/containers_and_restrictions.rb [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+              end
+            end
+          top_container =
+            # adding this failsafe after it hiccupped, though I don't fully understand why it's needed
+            unless top_container_record.nil?
+              "#{top_container_record['type']} #{top_container_record['indicator']}"
+            else ""
+            end
+          sub_container =
+            if instance.dig('sub_container').nil? == false
```
Comment (carolyncole) [inline reports/plain_vanilla_report_templates/containers_and_restrictions.rb:66]: @regineheberlein I prefer `! instance.dig('sub_container').nil?` vs `instance.dig('sub_container').nil? == false` It read for me as "not subcontainer is nil" vs "subconatiner is nil is false". It really is the same thing, but just a style perspective.

File: reports/plain_vanilla_report_templates/containers_and_restrictions.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+              sub1 = "#{instance.dig('sub_container', 'type_2')} #{instance.dig('sub_container', 'indicator_2')}"
+              unless instance.dig('sub_container', 'type_3').nil?
+                sub2 = "#{instance.dig('sub_container', 'type_3')} #{instance.dig('sub_container', 'indicator_3')}"
+              end
+              "#{sub1} #{sub2}"
+            end
+          do_boolean = instance['instance_type'] == "digital_object"
+        end # get_ao['instances'].each
+        row << [uri, "#{ead_id ||= ref_id}", title, date, level, "#{if do_boolean then do_boolean else false end}",
+                "#{top_container unless top_container.nil?} #{sub_container unless sub_container.nil?}"]
```
Comment (carolyncole) [inline reports/plain_vanilla_report_templates/containers_and_restrictions.rb:76]: do you need these unless in the string interpolation??  top_container for instance is always set to a string above.

if you need them, you could use the OR syntax `#{sub_container || ""}` which pass empty string to the interpolation is subcontainer is nil.

**Review by carolyncole — CHANGES_REQUESTED** [review]
Just a couple style questions.  Sorry this took me so long...

File: reports/plain_vanilla_report_templates/containers_and_restrictions.rb (hunk 1 above)
Comment (regineheberlein) [reply reports/plain_vanilla_report_templates/containers_and_restrictions.rb:66]: I don't know why, but making that change breaks it! All of a sudden it thinks it has to many "end"'s. Surely something to do with me structuring the code poorly, but I can't figure it out. For the moment, I'm changing it back to the double negative so I can get a report out the door...

**Pull request merged**